lib/engine/step/tracker.rb

```diff
@@ -71,14 +71,22 @@ def lay_tile(action, extra_cost: 0, entity: nil)
         discount = 0
 
         entity.abilities(:tile_lay) do |ability|
-          next if !ability.hexes.include?(hex.id) || !ability.tiles.include?(tile.name)
+          next unless ability.hexes.empty?
+
+          corporation = entity.owner
+          @game.game_error("Track laid in must be connected to one of #{corporation.id}'s stations") if
+            !@game.graph.reachable_hexes(corporation)[hex] && hex.name != corporation.coordinates
```

Comment:
will this break 1889? is this rule actually generic, also you shouldn't do reachable checks outside of a game.loading for performance reasons

Reply:
The only tile_lay ability outside 18MS with no hexes specified is NWR in 1882. The 1889 one have hexes as it specifies the coast towns.
Now I suppose 1882 should work anyway, but I have added 
  ability.when == 'track' &&
which excludes that ability.

But does not 1882 NWR require the tile lay to be connected?

An alternative is to add an attribute that specifies if tile lay need to be connected and use that to decide if this check should be done or not.